File: app/src/main/java/com/lostincontext/data/playlist/PlaylistPicker.java
package com.lostincontext.data.playlist;


import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.IOException;

public class PlaylistPicker {


    private Playlist playlist;

    public PlaylistPicker() {}

    public PlaylistPicker(Playlist playlist) {
        this.playlist = playlist;
    }

    public Playlist getPlaylist() {
        return playlist;
    }

    public void setPlaylist(Playlist playlist) {
        this.playlist = playlist;
    }

    //region Serializer / deserializer

    public String serialize() throws JsonProcessingException {
        return new ObjectMapper().writeValueAsString(this);

    }

    public static PlaylistPicker deserialize(String json) throws IOException {
        return new ObjectMapper().readerFor(PlaylistPicker.class).readValue(json);
    }
    //endregion
}

File: app/src/main/java/com/lostincontext/data/playlist/PlaylistPicker.kt
package com.lostincontext.data.playlist


import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper

import java.io.IOException

class PlaylistPicker {


    var playlist: Playlist? = null

    constructor() {
    }

    constructor(playlist: Playlist) {
        this.playlist = playlist
    }

    //region Serializer / deserializer

    @Throws(JsonProcessingException::class)
    fun serialize(): String {
        return ObjectMapper().writeValueAsString(this)

    }

    companion object {

        @Throws(IOException::class)
        fun deserialize(json: String): PlaylistPicker {
            return ObjectMapper().readerFor(PlaylistPicker::class.java).readValue<PlaylistPicker>(
                    json)
        }
    }
    //endregion
}
